src/plugins/intel_cpu/src/transformations/snippets/x64/pass/enforce_precision.cpp

```diff
@@ -125,5 +125,8 @@ std::set<std::vector<ov::element::Type>> EnforcePrecision::get_supported_precisi
     if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16) && ov::is_type<snippets::op::Brgemm>(op)) {
         return {{element::bf16, element::bf16}};
     }
+    if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx_fp16) && ov::is_type<snippets::op::Brgemm>(op)) {
+        return {{element::f16, element::f16}};
+    }
```

Comment:
I just think that the function body should be so:
```
std::set<std::vector<ov::element::Type>> types;
if (ov::is_type<snippets::op::Brgemm>(op)) {
    if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx_fp16))
         types.insert({element::f16, element::f16});
     if (dnnl::impl::cpu::x64::avx512_core_bf16)
         types.insert({element::bf16, element::bf16});
}
return types;
```

Also tell me please, if you have verified this pass and it works?

Reply:
I find we set MHA input and output type as infer precision in mha base test class: https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/shared/src/snippets/mha.cpp#L56
, so we still have full f16 model when we set input precision as f32 and f16 infer precision case. Is this intentional?

Reply:
update:
If I uncomment ["inType = outType = prc;](https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/shared/src/snippets/mha.cpp#L56)". Then the test model has f32 inputs and f16 infer_precision. In preLPT transformation, if infer_precision is f16, there will be a ConvertPrecision pass registered which insert convert(f32->f16) ops on full mode scope. The tokenized MHA still has f16 inputs from those convert ops output. Seems like bf16 and f16 infer_precision works differently. Enforce precision in MHA for f16 that insert convert will not happen, convert inserted in preLPT instead. I attached graphs in ticket.

Reply:
> Seems like bf16 and f16 infer_precision works differently.

You're totally right. If I remember correctly,
- if infer precision is `bf16`, we mark the nodes on `CPUGraph` using `EnforcePrecision`.
- if infer precision is `fp16`, we mark the nodes on `ov::Model` using `ConvertPrecision`.

But I think that anyway we should be sure that with infer precision `fp16`, Brgemms will be executed in this element type. So the current logic covers this case.